scripts/check_catalog_urls.py

```
@@ -0,0 +1,187 @@
+
+def check_url(url, timeout=20, max_retries=4):
+    """Return (status_code, error_message). Code 0 means network failure."""
+    # Reject anything that isn't HTTPS *before* handing it to urlopen,
+    # which otherwise happily opens file://, ftp://, custom schemes, etc.
+    # A broken or malicious catalog entry could otherwise probe the local
+    # filesystem or internal services.
+    parsed = urllib.parse.urlparse(url)
+    if parsed.scheme.lower() not in ALLOWED_SCHEMES:
+        return 0, f"unsupported scheme '{parsed.scheme}' (only https allowed)"
+    req = urllib.request.Request(safe_url(url), method="HEAD",
+                                 headers={"User-Agent": "AAVA-catalog-url-check/1.0"})
```

Ruff (0.15.11)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)